### PR TITLE
Fix port allocate iptables regex

### PR DESCRIPTION
**Pull request by tces1**
Fix issues: Conflicts arise when multiple containers use -p :xxx to automatically assign ports #2537

### CODE REVIEW

**Comment by tces1** [comment]
The old function cannot read "--dports 49162,49163,49164"

File: pkg/portutil/iptable/iptables.go [hunk 1]
```diff
@@ -27,16 +28,18 @@ func ParseIPTableRules(rules []string) []uint64 {
...
-	dportRegex := regexp.MustCompile(`--dports (\d+)`)
+	dportRegex := regexp.MustCompile(`--dports ((,?\d+)+)`)
 
 	for _, rule := range rules {
 		matches := dportRegex.FindStringSubmatch(rule)
 		if len(matches) > 1 {
-			port64, err := strconv.ParseUint(matches[1], 10, 16)
-			if err != nil {
-				continue
+			for _, matche := range strings.Split(matches[1], ",") {
```
Comment (AkihiroSuda) [inline pkg/portutil/iptable/iptables.go:36]:
```suggestion
			for _, match := range strings.Split(matches[1], ",") {
```

**Comment by tces1** [comment]
I have updated code, but two chesks fail.  It passed yesterday
I found the same issue in the check of [Merge pull request](https://github.com/containerd/nerdctl/commit/21ac3a5b76f03195fcf1220d8680d3c86b47003d) https://github.com/containerd/nerdctl/pull/2530 [from henry118/zstd](https://github.com/containerd/nerdctl/commit/21ac3a5b76f03195fcf1220d8680d3c86b47003d)

Can you help to analysis this issue?

**Comment by AkihiroSuda** [comment]
> I have updated code, but two chesks fail. It passed yesterday I found the same issue in the check of [Merge pull request](https://github.com/containerd/nerdctl/commit/21ac3a5b76f03195fcf1220d8680d3c86b47003d) #2530 [from henry118/zstd](https://github.com/containerd/nerdctl/commit/21ac3a5b76f03195fcf1220d8680d3c86b47003d)
> 
> Can you help to analysis this issue?

- https://github.com/containerd/containerd/issues/9163

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**